internal/cmd/commands/connect/funcs.go

```diff
@@ -99,15 +100,60 @@ func fmtSecretForTable(indent int, sc *targets.SessionCredential) []string {
 	if err != nil {
 		return origSecret
 	}
-	dst := new(bytes.Buffer)
-	if err := json.Indent(dst, in, fmt.Sprintf("%s    ", prefixStr), fmt.Sprintf("%s  ", prefixStr)); err != nil {
+
+	// The following operations are performed to better format unspecified credential types.
+	// Credential values are printed with no indentation so that X.509 certificates can be
+	// easily copied and pasted by the user.
+	var out []string
+	var baseCredMap map[string]any
+	err = json.Unmarshal(in, &baseCredMap)
+	if err != nil {
 		return origSecret
 	}
-	secretStr := strings.Split(dst.String(), "\n")
-	if len(secretStr) > 0 {
-		secretStr[0] = fmt.Sprintf("%s    %s", prefixStr, secretStr[0])
+	for k, iface := range baseCredMap {
+		switch iface := iface.(type) {
+		case nil:
+			out = append(out, fmt.Sprintf("%s    %s:", prefixStr, k))
+		case map[string]any:
+			// TODO: check if this is the only possible case for an 'unspecified' type credential
+			// and get rid of the other cases if this is so.
+			out = append(out, fmt.Sprintf("%s    %s:", prefixStr, k))
+			certs := make(map[string]any, len(iface))
+			for kk, vv := range iface {
+				if p, _ := pem.Decode([]byte(fmt.Sprintf("%v", vv))); p != nil {
```

Comment:
I'd sooner we have another type assertion than this string formatting, to make it a bit more predictable
```go
switch iface := iface.(type) {
case []byte:
	if p, _ := pem.Decode(iface); p != nil {
```
etc. Maybe one for `string` too. What do you think?

Reply:
would you expect a case where the unmarshalled interface is anything other than a []byte or a string? What would you recommend I do in that scenario? Does it sound reasonable to return an error in that case?